File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/ParseStr.java
package com.onevgo.functions;

import org.apache.http.NameValuePair;
import org.apache.http.client.utils.URLEncodedUtils;

import java.nio.charset.StandardCharsets;
import java.util.List;

public class ParseStr {
    public static List<NameValuePair> parseStr(String str) {
        return URLEncodedUtils.parse(str, StandardCharsets.UTF_8);
    }

    public static void main(String[] args) {
        String s = "first=value&arr[]=foo+bar&arr[]=baz";
        System.out.println(parseStr(s));
    }
}

File: 01_Language/01_Functions/php/parse_str.php
<?php

$str = "first=value&arr[]=foo+bar&arr[]=baz";

// Recommended
parse_str($str, $output);
echo $output['first'];  // value
echo $output['arr'][0]; // foo bar
echo $output['arr'][1]; // baz

// DISCOURAGED
//parse_str($str);
//echo $first;  // value
//echo $arr[0]; // foo bar
//echo $arr[1]; // baz
